databricks/koalas/series.py

```diff
@@ -4730,6 +4730,31 @@ def asof(self, where):
         result_series.name = self.name
         return result_series
 
+    def mad(self):
+        """
+        Return the mean absolute deviation of values.
+
+        Examples
+        --------
+        >>> s = ks.Series([1, 2, 3, 4])
+        >>> s
+        0    1
+        1    2
+        2    3
+        3    4
+        Name: 0, dtype: int64
+
+        >>> s.mad()
+        1.0
+        """
+
+        sdf = self._internal.spark_frame
+        column_name = self._internal.data_spark_column_names[0]
+        avg = _unpack_scalar(sdf.select(F.avg(sdf[column_name])))
+        mad = _unpack_scalar(sdf.select(F.avg(F.abs(sdf[column_name] - avg))))
```

Comment:
You don't have to run the jobs twice here. Can we do this in single pass? e.g.,  `return _unpack_scalar(sdf.select(F.avg(F.abs(sdf[column_name] - F.avg(sdf[column_name])))))`

Reply:
I think the proposed method may not work properly because the aggregate function `avg` is used  as an argument of another aggregate function `avg`.  
I can't find any other way except running the jobs twice at this time. 
Are there any other ideas can handle it by running a job once? 
I would appreciate any comments :)

Reply:
Okay.